dashboard/app/views/home/_homepage.html.haml

```diff
@@ -14,19 +14,37 @@
     - homepage_data[:isTeacher] = true
     - homepage_data[:joined_sections] = @student_sections
     - homepage_data[:announcement] = DCDO.get('announcement_override', nil)
-    - show_census_banner = !!(current_user.show_census_teacher_banner?)
+    - if !current_user.donor_teacher_banner_dismissed
+      - donor_banner_name = current_user.donor_teacher_banner_name
+    - else
+      - donor_banner_name = @force_donor_teacher_banner
+    - show_census_banner = !!(!donor_banner_name && current_user.show_census_teacher_banner?)
     - homepage_data[:showCensusBanner] = show_census_banner
     - homepage_data[:hiddenScripts] = current_user.get_hidden_script_ids
+    - homepage_data[:donorBannerName] = donor_banner_name
     - if show_census_banner
       %script{type: "text/javascript", src: "https://maps.googleapis.com/maps/api/js?client=#{CDO.google_maps_client_id}&sensor=true&libraries=places,geometry&v=3.7"}
       - teachers_school = current_user.school_info.school
       - school_stats = SchoolStatsByYear.where(school_id: teachers_school.id).order(school_year: :desc).first
       - homepage_data[:ncesSchoolId] = teachers_school.id
       - homepage_data[:censusQuestion] = school_stats.try(:has_high_school_grades?) ? "how_many_20_hours" : "how_many_10_hours"
+      - homepage_data[:currentSchoolYear] = current_census_year
       - homepage_data[:teacherName] = current_user.name
       - homepage_data[:teacherId] = current_user.id
       - homepage_data[:teacherEmail] = current_user.email
-      - homepage_data[:currentSchoolYear] = current_census_year
+    - elsif donor_banner_name
+      - homepage_data[:teacherFirstName] = current_user.short_name
+      - homepage_data[:teacherSecondName] = current_user.second_name
+      - homepage_data[:teacherId] = current_user.id
+      - homepage_data[:teacherEmail] = current_user.email
```

Comment:
Intentional that we rewrite `teacherEmail` here when it's defined on line 34 above?

Reply:
As of the latest commit, we run one or the other.